parakeet_rocm/models/parakeet.py

```
@@ -109,35 +121,53 @@ def get_model(model_name: str = PARAKEET_MODEL_NAME) -> ASRModel:
+    with _cache_lock:
+        # Note: currsize > 0 does not guarantee *this* model_name is cached.
+        # If the cache contains a different model, _get_cached_model(model_name)
+        # will trigger a load — contradicting the "no-load" guarantee.
+        # With maxsize=4 and ≤2 model names in practice, this is safe.
+        # Edge case documented in module docstring.
+        if _get_cached_model.cache_info().currsize == 0:  # type: ignore[attr-defined]
+            return
         try:
-            torch.cuda.empty_cache()
+            model = _get_cached_model(model_name)
         except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)